### DISCUSSION TITLE
What happened with spindle pwm?

### DISCUSSION THREAD

**Discussion by fitch22** [grblHAL/core#94] [Q&A]
OK, I will admit I have not built grblHal in a while.  Today I updated my repository, but also STM32 Cube got updated as well.  I built new firmware.  Was looking at the PWM output and needed to change the frequency as I have a pwm to dc board that only works up to 3KHz.  So I think I accidentally set $31 to 12000 and I should have set $30 to 12000.  Anyway, PWM output was working, but when I try to change $31 and $30 to 0 and 12000, I get error:53.  I power cycled, no change.  Now it looks like all spindle related options are not showing up at all, even with $+.  ioSender shows no spindle related settings.

Any idea what happened?  How do I get out of this, or do I need to reset the flash?

Thanks.

**Comment by terjeio** [comment]
Which board map? 
Can you check _driver.c_ after an _Index > Rebuild_ to see if `hal.driver_cap.variable_spindle = On;` is compiled in, I get:

![image](https://user-images.githubusercontent.com/20260062/144331717-2d02cf23-50b4-4abe-a973-2bf63046f97c.png)

If not the PWM settings are not available reulting in error 53.

**Reply by fitch22** [reply]
Hi Terje,
Board map is btt_skr_2.0_map.h.
My driver.c looks just like yours.

So, I solved it by erasing the flash and reprogramming.  I am now able to see all the spindle option settings once again and can change the pwm frequency and duty cycle.

Could this be because the flash layout changed?

Thanks,
Bill

**Reply by terjeio** [reply]
> Could this be because the flash layout changed?

No - I do not think so. Corrupted RAM? There is a check for variable spindle capability [here](https://github.com/grblHAL/core/blob/637fa80eb328b042754716969674caff598425d9/settings.c#L1412-L1419).
